src/components/Modals/CitizenRepealLegislationModal.js

```diff
@@ -32,94 +37,101 @@ function CitizenRepealLegislationModal({
     discussionDescription,
     discussionLink,
   }) => {
-    dispatch(democracyActions.citizenProposeRepealLegislation.call({
-      discussionName,
-      discussionDescription,
-      discussionLink,
-      tier,
-      id,
-      section,
-    }));
+    dispatch(
+      democracyActions.citizenProposeRepealLegislation.call({
+        discussionName,
+        discussionDescription,
+        discussionLink,
+        tier,
+        id,
+        section,
+      }),
+    );
     closeModal();
   };
 
+  const { dialogStep, handleClick } = useAgreeDisagreeModal(isValid, trigger);
+
   return (
     <form
       className={styles.getCitizenshipModal}
       onSubmit={handleSubmit(onSubmitRepeal)}
     >
-      <div className={styles.h3}>Propose referendum for legislation repeal</div>
+      {dialogStep === 'form' ? (
```

Comment:
I dodnt test, but wont this cause whole form im the back to disappear from behind the modal? If yes then I dont like it, it will cause jarring jumps.